clang/lib/Driver/ToolChains/DPURTE.cpp

```diff
@@ -153,6 +153,7 @@ void Linker::ConstructJob(Compilation &C, const JobAction &JA,
     CmdArgs.push_back("-T");
     CmdArgs.push_back(TCArgs.MakeArgString(sysroot + "/link/dpu.lds"));
 
+#define DPU_NR_THREADS DPU_NR_THREADS_V1A
```

Comment:
I assume you chose that because DPU_NR_THREADS_V1A > DPU_NR_THREADS_V1B.
Consider either
#define DPU_NR_THREADS (DPU_NR_THREADS_V1A > DPU_NR_THREADS_V1B ? DPU_NR_THREADS_V1A : DPU_NR_THREADS_V1B)
(should evaluate in the preprocessor)
or
constexpr unsigned int DPU_NR_THREADS = std::max(DPU_NR_THREADS_V1A, DPU_NR_THREADS_V1B);
In the worst case, adding a comment to always pick the larger value.